packages/paste-core/components/toast/src/useToaster.ts

```diff
@@ -3,21 +3,51 @@ import {uid} from '@twilio-paste/uid-library';
 import type {ToasterToast, ToasterPush, UseToasterReturnedProps} from './types';
 
 export const useToaster = (): UseToasterReturnedProps => {
+  const isMounted = React.useRef<boolean | null>(null);
   const [toasts, setToasts] = React.useState<ToasterToast[]>([]);
 
+  React.useEffect(() => {
+    isMounted.current = true;
+    return () => {
+      isMounted.current = false;
+      /**
+       * Clear all timeouts from the toaster when the component unmounts
+       */
+      toasts.forEach((toast) => {
+        if (toast.timeOutId) {
+          window.clearTimeout(toast.timeOutId);
+        }
+      });
+    };
+  }, []);
+
   const pop = (id: ToasterToast['id']): void => {
+    if (!isMounted.current) {
+      return;
+    }
```

Comment:
How likely is this to happen? Method called when unmounted?

Reply:
It was happening in the tests. I somehow was calling `pop` after the component had been unmounted, I think by holding a reference to it. The added safety doesn't cost much, as this isn't hot code.

Reply:
👍🏼